llvm-spirv/test/transcoding/SPV_INTEL_function_pointers/fp-in-recusive-type.ll

```diff
@@ -0,0 +1,84 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_function_pointers -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.spv -r -o %t.r.bc
+; RUN: llvm-dis %t.r.bc -o - | FileCheck %s --check-prefix=CHECK-LLVM
+;
+; This test is intended to check the following source code which caused
+; assertion failure and subsequent crash:
+;
+; virtual SPIRV::SPIRVEntry* SPIRV::SPIRVModuleImpl::getEntry(SPIRV::SPIRVId) const: Assertion `Loc != IdEntryMap.end() && "Id is not in map"' failed.
+;
+; typedef void(*fp_t)(struct Desc *);
+;
+; struct Type;
+;
+; struct Desc {
+;   int B;
+;   Type *Ptr;
+; };
+;
+; struct Type {
+;   int A;
+;   fp_t FP;
+; };
+;
+; Type T;
```

Comment:
Probably it would be more clear if you show how this is used in the kernel to better match with LLVM IR below.

Reply:
Done